app/workers/send_eoc_deadline_reminder_email_to_candidates_worker.rb

```diff
@@ -2,24 +2,18 @@ class SendEocDeadlineReminderEmailToCandidatesWorker
   include Sidekiq::Worker
 
   def perform
-    applications_to_send_reminders_to.each do |application|
-      SendEocDeadlineReminderEmailToCandidate.call(application_form: application)
+    if applications_to_send_reminders_to.present?
+      applications_to_send_reminders_to.find_each(batch_size: 100) do |application|
+        SendEocDeadlineReminderEmailToCandidate.call(application_form: application)
+      end
```

Comment:
without wrapping in the conditional checking for presence this test blows up 

![image](https://user-images.githubusercontent.com/42515961/128163513-97af9327-4943-4d04-a838-ddce0f2f4804.png)


any ideas?

Reply:
I think in your test you need to set it up so that your db records have type `ApplicationForm::ActiveRecord_Relation` rather than 'Array`

Reply:
Actually it may be because you return an array if it's not time to send the email?
https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/workers/send_eoc_deadline_reminder_email_to_candidates_worker.rb#L11

Reply:
Looks like you should do an early return in the perform method (ie add a guard clause if its not time to send the email)

Reply:
Yea it still blows up with a guard clause unfortunately